libcextract/FunctionDepsFinder.hh

```diff
@@ -43,6 +43,33 @@ class ClosureSet
     Dependencies.insert(decl);
   }
 
+  /*
+   * Check if the same Decl was already added into the closure but for a
+   * different typename.
+   */
+  TypedefDecl *insideRangeOfDecl(TypedefDecl *decl)
+  {
+    SourceRange range = decl->getSourceRange();
+
+    for (auto it = Dependencies.begin(); it != Dependencies.end(); ++it) {
+      if (TypedefDecl *cur_decl = dynamic_cast<TypedefDecl *>(*it)) {
+        SourceRange cur_range = cur_decl->getSourceRange();
+
+        // Avoid comparing to itself
+        if (cur_range == range)
+          continue;
+
+        if (cur_range.getBegin() == range.getBegin() &&
```

Comment:
You can use Contains_From_LineCol and get the Decl with larger Range.

Reply:
Cool, I'll use it